kafka-connect-mqtt/src/main/scala/com/datamountaineer/streamreactor/connect/mqtt/source/MqttMsgKey.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * *
+ *   * Copyright 2016 Datamountaineer.
+ *   *
+ *   * Licensed under the Apache License, Version 2.0 (the "License");
+ *   * you may not use this file except in compliance with the License.
+ *   * You may obtain a copy of the License at
+ *   *
+ *   * http://www.apache.org/licenses/LICENSE-2.0
+ *   *
+ *   * Unless required by applicable law or agreed to in writing, software
+ *   * distributed under the License is distributed on an "AS IS" BASIS,
+ *   * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   * See the License for the specific language governing permissions and
+ *   * limitations under the License.
+ *   *
+ */
+
+package com.datamountaineer.streamreactor.connect.mqtt.source
+
+import com.sksamuel.avro4s.{RecordFormat, SchemaFor}
+import io.confluent.connect.avro.AvroData
+
+case class MqttMsgKey(topic: String, id: Int)
+
+object MqttMsgKey {
+  private val recordFormat = RecordFormat[MqttMsgKey]
+  private val avroSchema = SchemaFor[MqttMsgKey]()
+  private val avroData = new AvroData(1)
```

Comment:
This converter will always deal with one schema generated by the MqttMsgKey case class. therefore we only allocate 1 element in the cache array within avrodata